### DISCUSSION TITLE
Pulumi native provider generated code for secret value outputs bug

### DISCUSSION THREAD

**Discussion by frezbo** [pulumi/pulumi#8965] [Pulumi Providers] [answered]
When generating a secret value as output using a sample schema like https://github.com/frezbo/pulumi-provider-talos/blob/8300bfa1ea9a42e2bd9748a4ccf8d36d7af7ec54/provider/cmd/pulumi-resource-talos/schema.json#L124-L157 if output property `secretsBundle` is not duplicated in `inputProperties` and the output property is marked as a `secret` by adding like below snippet

```json
"properties": {
                "secretsBundle": {
                    "type": "object",
                    "$ref": "#types/talos:bundle:SecretsBundle",
                    "secret": true
                }
            },
```

The generated code is like this:

```go
+       if args.SecretsBundle != nil {
+               args.SecretsBundle = pulumi.ToSecret(args.SecretsBundle).(SecretsBundleTypeOutput)
+       }
+       secrets := pulumi.AdditionalSecretOutputs([]string{
+               "secretsBundle",
+       })
+       opts = append(opts, secrets)
```

as seen from the generated code, it considers `SecretsBundle` as an input argument as `args.SecretsBundle`

I've also looked into the pulumi code and I believe https://github.com/pulumi/pulumi/pull/7128 this change might have broken it. The test for the provider codegen https://github.com/pulumi/pulumi/blob/a1e18dae4dc00142c086fcae0a90171f278db166/pkg/codegen/testing/test/testdata/simple-yaml-schema/schema.yaml#L77-L85 also defines an input property and an output property with the same name `bar`. Is that a requirement for the provider? I can't think of a use case where the input and output property always need to be the  same name.

**Comment by nimbinatus** [comment, marked as answer]
Just in case it's not clear from the GitHub UI, we've opened an issue to correspond to this question. Thanks for bringing it up! I'll leave this open until there's more information in the attached issue: https://github.com/pulumi/pulumi/issues/8967

**Reply by frezbo** [reply]
Fixed as of https://github.com/pulumi/pulumi/pull/9052
